src/psyclone/psyGen.py

```diff
@@ -1130,13 +1140,16 @@ def reprod_reduction(self):
 
     @property
     def local_reduction_name(self):
-        '''Generate a local variable name that is unique for the current
+        '''
+        Generate a local variable name that is unique for the current
         reduction argument name. This is used for thread-local
-        reductions with reproducible reductions '''
-        tag = self._reduction_arg.name
-        name = self.ancestor(InvokeSchedule).symbol_table.\
-            find_or_create_tag(tag, "l_" + tag).name
-        return name
+        reductions with reproducible reductions.
+
+        :returns: local reduction variable name.
+        :rtype: str
+
+        '''
+        return "l_" + self.reduction_arg.name
```

Comment:
Any reason you removed the symbol_table operation? If you know for sure that the symbol already exist you can use
`...symbol_table.lookup_with_tag(tag).name` so if there is a symbol with a clashing name on the original code this would resolve it. If the symbol may not exist we still need the `find_or_create_tag`

Reply:
I removed the operation of creating a symbol from here because the symbol for local reduction is only required if we are doing reproducible reductions. Otherwise the existing symbol associated with `reproduction_arg` is enough. Besides, the removed code here did not generate the correct data symbol for an array of dimensions `(1, omp_thread_index)`.

I moved the symbol generation to `if self.reprod_reduction` branch of `_reduction_reference` function, as we need to return one Reference from two options: a single-valued argument for non-reproducible reductions and an array argument for reproducible reductions. The array data symbol is now created in `reduction_reference` for reproducible reactions, i.e. `symtab.find_or_create_tag(tag=self.local_reduction_name, ...)`.

I appreciate you would like to check for a possible clash, but then I think creating the symbol here only makes sense if it is done under `if self.reprod_reduction` condition, and it is not quite clear to me what the property would return if that condition is not met. I am refactoring global reduction infrastructure anyway, so we can perhaps decide on this there?

Reply:
I don't feel too good about how this is implemented. For example I would expect that a method called *_name returns the name of the symbol but it now returns the tag (the symbol name may actually be different after doing the lookup).

And in general I think generation of systems should happen earlier, at psy-layer generation time, or at transformation time if it comes from a modification introduced by applying a transformation. The node methods should only retrieve information already existing on the tree.

However, I do agree that fixing this seem out of scope for this PR. And since the CI run shows we are getting correct results for LFRic, and I am happy to leave this for a future PR/discussion.

Reply:
That is a valuable feedback on implementation, thank you! I linked your comments in #2381 and I am happy to add the relevant TODO to this issue to make sure this is dealt with.

Reply:
Added TODOs in `local_reduction_name` and `_reduction_reference`.